hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -804,15 +833,45 @@ public long getScmTerm() throws NotLeaderException {
    * Notify ReplicationManager that the command counts on a datanode have been
    * updated via a heartbeat received. This will allow RM to consider the node
    * for container operations if it was previously excluded due to load.
-   * @param datanodeDetails The datanode for which the commands have been
-   *                        updated.
+   * @param datanode The datanode for which the commands have been updated.
    */
-  public void datanodeCommandCountUpdated(DatanodeDetails datanodeDetails) {
-    // For now this is a NOOP, as the plan is to use this notification in a
-    // future change to limit the number of commands scheduled against a DN by
-    // RM.
+  public void datanodeCommandCountUpdated(DatanodeDetails datanode) {
     LOG.debug("Received a notification that the DN command count " +
-        "has been updated for {}", datanodeDetails);
+        "has been updated for {}", datanode);
+    excludedNodes.compute(datanode, (k, v) -> {
+      if (v != null) {
+        // There is an existing mapping, so we may need to remove it
+        try {
+          if (getDatanodeQueuedReplicationCount(datanode)
+              < datanodeReplicationLimit) {
+            // Returning null removes the entry from the map
+            return null;
+          } else {
+            return 1;
+          }
+        } catch (NodeNotFoundException e) {
+          LOG.warn("Unable to find datanode {} in nodeManager. " +
+              "Should not happen.", datanode);
+          return null;
+        }
+      }
+      // If there is no existing mapping, then return null as we don't want to
+      // add a mapping.
+      return null;
+    });
```

Comment:
Can be simplified a bit by using `computeIfPresent` instead of `compute`:

```suggestion
    // If there is an existing mapping, we may need to remove it
    excludedNodes.computeIfPresent(datanode, (k, v) -> {
      try {
        if (getDatanodeQueuedReplicationCount(datanode)
            < datanodeReplicationLimit) {
          // Returning null removes the entry from the map
          return null;
        } else {
          return 1;
        }
      } catch (NodeNotFoundException e) {
        LOG.warn("Unable to find datanode {} in nodeManager. " +
            "Should not happen.", datanode);
        return null;
      }
    });
```